examples/mass_delete_old_medias.py

```diff
@@ -0,0 +1,460 @@
+#!/usr/bin/env python3
+"""
+Script which aims at freeing storage by deleting medias older than a
+given number of days. Depending on the parameters given, the script
+will either delete the medias or email speakers about the impending
+deletion, to give them time to protect their medias by applying a
+category to them.
+"""
+
+import argparse
+from datetime import date, datetime
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+import getpass
+import logging
+import os
+from pathlib import Path
+import smtplib
+import ssl
+import sys
+from typing import Optional
+
+try:
+    from ms_client.client import MediaServerClient
+except ModuleNotFoundError:
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    from ms_client.client import MediaServerClient
+
+
+logger = logging.getLogger(__name__)
+DEFAULT_PLAIN_EMAIL_TEMPLATE = (
+    'The following {media_count} media should be deleted according to '
+    'the University policy. You have until {delete_date} to review '
+    'each media below and set the "{skip_category}" category if you '
+    'need to preserve this content. After this date, the media will be '
+    'deleted. You can also delete it when you review it.\n\n'
+    'List of media to review ({media_count}):\n'
+    '{list_of_media}\n'
+)
+DEFAULT_HTML_EMAIL_TEMPLATE = (
+    '<p>The following {media_count} media should be deleted according '
+    'to the University policy. You have until {delete_date} to review '
+    'each media below and set the "{skip_category}" category if you '
+    'need to preserve this content. After this date, the media will be '
+    'deleted. You can also delete it when you review it.</p>\n'
+    '<br/>\n'
+    '<p>List of media to review ({media_count}):</p>\n'
+    '<ul>{list_of_media}</ul>\n'
+)
+
+
+def pp_size(size_bytes: int) -> str:
+    """
+    Return human-readable size with automatic suffix.
+    """
+    for unit in ('', 'K', 'M', 'G', 'T', 'P', 'E', 'Z'):
+        if abs(size_bytes) < 1000:
+            return f'{size_bytes:.1f}{unit}B'
+        size_bytes /= 1000
+    return f'{size_bytes:.1f}YB'
+
+
+def _get_old_medias(
+    msc: MediaServerClient,
+    before_date: date,
+    skip_category: str
+) -> list[dict]:
+    response = msc.api('catalog/get-all/', params={'format': 'flat', 'timings': 'yes'})
+    skip_category = skip_category.lower()
+    old_medias = []
+    for key in ('videos', 'lives', 'photos'):
+        medias = response.get(key, ())
+        for media in medias:
+            add_date = datetime.strptime(media['add_date'], '%Y-%m-%d %H:%M:%S').date()
+            categories = {cat.strip(' \r\t').lower() for cat in (media['categories'] or '').split('\n')}
+            if add_date < before_date and (not skip_category or skip_category not in categories):
+                old_medias.append(media)
```

Comment:
in --dry-run mode print why a media is skipped (because of the date  or of the category)